modin/engines/base/frame/data.py

```diff
@@ -211,7 +211,7 @@ def _set_columns(self, new_columns):
                 self._dtypes.index = new_columns
         self._apply_index_objs(axis=1)
 
-    def _set_axis(self, axis, new_axis):
+    def _set_axis(self, axis, new_axis, cache_only=False):
```

Comment:
Why would we want external and internal indices to be mismatched?

Reply:
that could happen in case of incorrect usage of reduce functions, if we expect an output as `Series`, so we set columns like `["__reduced__"]`, but the actual result is `DataFrame` so the lengths of internal and external indices do not matches.

We can try to update external indices or could raise an exception. What do you think we should do in that case?

Reply:
Do you have a code example I can use (or point to where the error shows)? I am still not sure I understand

Reply:
Let's say we use some of reduce function at modin frame (for example [`_fold_reduce`](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/data.py#L959-L973) ) that expects the resulted frame to be Series-like, because it sets `new_columns` as `["__reduced__"]` in [`_compute_map_reduce_metadata`](https://github.com/modin-project/modin/blob/9ed8e06960cb610680763039a44ae211ea82931c/modin/engines/base/frame/data.py#L930-L957). However we can't guarantee that the result of `_fold_reduce` is always a `Series`, it's just a convention of using `_fold_reduce`. So, if the result of `_fold_reduce` is `DataFrame` with more than 1 column we'll run into problems when we'll try to propagate `["__reduced__"]` columns into partitions because the lengths of axes will be mismatched. In that case, we may want to use internal indices as external to save consistency between our internal and external indices and do not raise an exception

Reply:
I've added `force` parameter at `_validate_axis_equality` to specify that behavior

Reply:
We have a utility for this already, can we use it instead of adding this here:

https://github.com/modin-project/modin/blob/9ed8e06960cb610680763039a44ae211ea82931c/modin/engines/base/frame/partition_manager.py#L402-L431

Reply:
Well, what `get_indices` is doing is just compute internal indices for modin frame and return it.
What `_validate_axis_equality` is doing is gets internal indices via `get_indices`, compares it with external indices and 

- propagate external indices into partitions if their not equals, but their lengths are equal
- raising an exception if their not equals by content and lengths and `force=False`
- sets external indices equal to internal if their not equals by content and lengths and `force=True` 

So it's already uses `get_indices`.